include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_dual_splitting.cpp

```diff
@@ -316,20 +316,30 @@ TimeIntBDFDualSplitting<dim, Number>::do_timestep_solve()
   pde_operator->interpolate_velocity_dirichlet_bc(velocity_dbc_np, this->get_next_time());
 
   // perform the sub-steps of the dual-splitting method
-  convective_step();
+  if(this->update_velocity)
+  {
+    convective_step();
+  }
 
-  pressure_step();
+  if(this->update_pressure)
+  {
+    pressure_step();
```

Comment:
Assuming we have `update_velocity == false` and `update_pressure == true` and that we do not solve for the ALE problem: What does the fluid time stepper actually solve for in this case? It solves the pressure but why does this actually change the solution? I understand that in this case we have a coupling "fluid -> structure" via the stress boundary condition (which is affected by the pressure solution), but what exactly is the coupling "structure -> fluid" in this case? Is it only the coupling via the pressure Neumann boundary condition that remains?

Before looking at this code, I would have expected that this UpdateMethod solves the convective step (which is explicit and which affects the pressure rhs-vector) as well, but then only skip the steps after the `pressure_step()` (which are implicit and which might be expensive). If we implement this variant as described here, it is important that we make sure that the `convective_step()` does not manipulate any velocity dof vectors apart from temporary vectors (or vectors of local scope). It could then potentially happen that we falsify the "true" velocity solution (which is "unconverged" in the FSI sense, but taking into account the viscous term and the projection onto divergence free vectors).

Reply:
yes, it just updates the pressure given new boundary data.
in the dual-splitting scheme, it enters via $h_p$:
pressure step:
![image](https://github.com/user-attachments/assets/4e30dbae-038f-4468-b43e-14a726146198)
with $h_p$ defined as:
![image](https://github.com/user-attachments/assets/0e97720f-b3ed-4c32-87d2-87cf8b3f8660)
(and where the read part should be considered when extending towards gen Newtonian fluids, coming soon!)

The semi-implicit versions are not unique in general,  but I wanted to get to the cheapest scheme possible.
The relevant BCs do capture the change of interface data (a change in the displacement "reaches" the PPE).
This is the case since `do_timestep_solve()` in any case involves
` pde_operator->interpolate_velocity_dirichlet_bc(velocity_dbc_np, this->get_next_time());`

I think we follow the credo 
"update_velocity == false means not touching the velocity fields"
we should not come into a situation, where we would have to execute the entire scheme to get a valid `velocity_np`. Vice versa, same holds true for the pressure!

So in general I would say i) we have to be careful not to leave solution vectors in an "invalid" state (meaning not considering for some terms) and ii) that we do not perform updates that are not necessary (for efficiency).
Or do you mean something different here? Am I overlooking something?